src/photos/ducks/clustering/albums.js

```diff
@@ -58,64 +73,64 @@ const addAutoAlbumReferences = async (photos, album) => {
       log('info', `Nothing to clusterize for ${album._id}`)
     }
   } catch (e) {
-    log('error', e.reason)
+    log('error', e)
   }
   return refCount
 }
 
-const createAutoAlbum = async photos => {
+const createAutoAlbum = async (client, photos) => {
   const name = albumName(photos)
   const period = albumPeriod(photos)
   const created_at = new Date()
   const album = { name, created_at, auto: true, period }
-  return cozyClient.data.create(DOCTYPE_ALBUMS, album)
+  const result = await client.create(DOCTYPE_ALBUMS, album)
+  const autoAlbum = client.hydrateDocument(result.data)
+  return autoAlbum
 }
 
-const removeAutoAlbums = async albums => {
+const removeAutoAlbums = async (client, albums) => {
   for (const album of albums) {
-    await cozyClient.data.delete(DOCTYPE_ALBUMS, album)
+    await client.destroy(album)
   }
 }
 
-const removeAutoAlbumReferences = async (photos, album) => {
-  await cozyClient.data.removeReferencedFiles(album, photos.map(p => p.id))
+const removeAutoAlbumReferences = async (client, photos, album) => {
+  const ids = []
   for (const photo of photos) {
+    ids.push(photo.id)
     photo.clusterId = ''
   }
+  await removeRefs(client, photos, album)
 }
 
-export const findAutoAlbums = async () => {
-  const autoAlbumsIndex = await cozyClient.data.defineIndex(DOCTYPE_ALBUMS, [
-    'auto',
-    'name'
-  ])
-
-  let next = true
-  let skip = 0
-  let autoAlbums = []
-  while (next) {
-    const results = await cozyClient.data.query(autoAlbumsIndex, {
-      selector: { auto: true },
-      sort: [{ name: 'desc' }],
-      skip: skip,
-      wholeResponse: true
-    })
-    if (results && results.docs) {
-      autoAlbums = autoAlbums.concat(results.docs)
-      skip = autoAlbums.length
-      if (!results.next) {
-        next = false
+export const findAutoAlbums = async client => {
+  const query = client
+    .find(DOCTYPE_ALBUMS)
+    .where({ auto: true })
+    .sortBy([
+      {
+        name: 'desc'
       }
-    }
+    ])
+
+  let result = await client.query(query)
+  let hydrated = client.hydrateDocuments(DOCTYPE_ALBUMS, result.data)
+  let autoAlbums = hydrated
+  let skip = autoAlbums.length
+  while (result.next) {
+    result = await client.query(query.offset(skip))
+    hydrated = client.hydrateDocuments(DOCTYPE_ALBUMS, result.data)
+    autoAlbums = autoAlbums.concat(hydrated)
+    skip = autoAlbums.length
   }
```

Comment:
We have a `queryAll` method now: https://github.com/cozy/cozy-client/commit/932cdc18ae18762eba90f229810f531563110f27#diff-405c53e28840707e47a25b26d2ed602bR445